### DISCUSSION TITLE
typing strategy for text & literal_column usage

### DISCUSSION THREAD

**Discussion by jvanasco** [sqlalchemy/sqlalchemy#11395] [Usage Questions]
My question is related to the recent #11379

I am typing a large legacy application that has a few "macros" for date operations. 

These range in complexity, but are all relatively simple:

    def sql_now() -> TextClause:
        return text("(CURRENT_TIMESTAMP AT TIME ZONE 'UTC')")

    def sql_now_minus_interval(
        interval: str = "2 MONTHS",
    ) -> TextClause:
        return text(
            "(CURRENT_TIMESTAMP AT TIME ZONE 'UTC' - INTERVAL :interval)"
        ).bindparams(interval=interval)

The typing issue I am facing, is this:

* Everything is using `text`/`TextClause`, because the usage within the typed package has been as part of larger column expressions.
* Running the mypy against some of the packages that use this code, I now have a lot of column usage -- e.g. `foo.created_at = sql_now()` - for which `literal_column` / `ColumnClause` is more appropriate [the #11379 ticket]

Does anyone have a good strategy that can be used for both contexts?  I would prefer not to have 2 identical functions, or pass in an argument just to use `@overload` to customize the return type.

**Comment by CaselIT** [comment]
Hi,

For the first one I think you can use the a literal column.
For the second one I think literal column doesn't support parameters, but maybe you can make sqlalchemy do the operation doing `sql_now() - text('INTERVAL ...')...`?
I'm not sure if the type returned by the second one works
